docs/notebooks/failure_ego.pct.py

```diff
@@ -117,26 +115,19 @@ def create_classification_model(data):
 # For our `GPR` model, we will use a standard L-BFGS optimizer from Scipy, whereas we will optimze our `VGP` model using alternate Adam steps (to optimize kernel parameter) and NatGrad steps (to optimize variational parameters).
 
 # %% [markdown]
-# We'll train the GPR model with an L-BFGS-based optimizer, and the GPC model with the custom algorithm above.
+# We'll train the GPR model with the default L-BFGS-based optimizer, and the GPC model with the custom algorithm above.
 
 # %%
-from trieste.models.gpflow import GPflowModelConfig
+from trieste.models.gpflow.models import GaussianProcessRegression, VariationalGaussianProcess
+from trieste.models.optimizer import BatchOptimizer
 
 models: dict[str, trieste.models.ModelSpec] = {
-    OBJECTIVE: GPflowModelConfig(**{
-        "model": regression_model,
-        "optimizer": gpflow.optimizers.Scipy(),
-    }),
-    FAILURE: GPflowModelConfig(**{
-        "model": classification_model,
-        "model_args": {
-            "use_natgrads": True,
-        },
-        "optimizer": tf.optimizers.Adam(1e-3),
-        "optimizer_args": {
-            "max_iter": 50,
-        },       
-    }),
+    OBJECTIVE: GaussianProcessRegression(regression_model),
+    FAILURE: VariationalGaussianProcess(
+        classification_model,
+        BatchOptimizer(tf.optimizers.Adam(1e-3), max_iter=50),  
```

Comment:
here we need to explicitly pass the `BatchOptimizer` object because we are using non-default settings (`max_iter=50`), otherwise the wrapper tells you which `Optimizer` you need to use

I think there is no way around it, as soon as they need a non-default optimizer they need to know about it
what we could do is the following:
1. Explicitly educate users in one of the intro notebooks on difference between `Optimizer` and `BatchOptimizer` and when to use them
2. Eliminate it from the rest of the notebooks and simply rely on defaults set in the interface, notebooks should work fine with the defaults

Reply:
If there is a nice default, i'd go with option 2.

Reply:
ah, these weren't meant to be options, I was proposing doing both things :)